packages/eslint-plugin/src/lib/rules/ban-types.ts

```diff
@@ -2,14 +2,18 @@
  * @fileoverview Enforces that types will not to be used
  * @author Armano <https://github.com/armano2>
  */
-'use strict';
 
-const util = require('../util');
+import { Rule } from 'eslint';
+import * as util from '../util';
 
 //------------------------------------------------------------------------------
 // Rule Definition
 //------------------------------------------------------------------------------
 
+interface TypeErrorConfig {
+  message: string;
+  fixWith?: string;
+}
 const defaultOptions = [
```

Comment:
instead of the cast below, linters for example generally prefer `const defaultOptions: Record<string, TypeErrorConfig> = {`

Reply:
Not quite correct, because this is an array with an object with a property of type `Record<string, TypeErrorConfig>`, but your idea is correct.
I was lazy here - it would have been better to type this at the variable level rather than at the property level.